File: C++/Array/Triplet Sum in Array.cpp
class Solution{
    public:
    //Function to find if there exists a triplet in the 
    //array A[] which sums up to X.
    bool find3Numbers(int A[], int n, int X)
    {
        //Your Code Here
        sort(A, A+n);
        for(int i = 0; i < n; i++){
            int y = X - A[i];
            int low = i+1, high = n-1;
            while (low < high){
                if(A[low] + A[high] == y){
                    return 1;
                }
                else if(A[low] + A[high] > y){
                    high--;
                }
                
                else if(A[low] + A[high] < y){
                    low++;
                }
                
            }
        
        }
        return 0;
    }

};

File: Python/Array/Triplet Sum in Array.py
class Solution:
     
    #Function to find if there exists a triplet in the 
    #array A[] which sums up to X.
    def find3Numbers(self,A, n, X):
        # Your Code Here
        A.sort()
        for i in range(n):
            y = X - A[i]
            low = i + 1
            high = n-1
            while low < high:
                if A[low] + A[high] == y:
                    return 1
                
                elif  A[low] + A[high] < y:
                    low += 1
                
                elif A[low] + A[high] > y:
                    high -= 1
        return 0   
